PermissionScope/PermissionScope.swift

```diff
@@ -37,11 +37,16 @@ public struct PermissionConfig {
     let type: PermissionType
     let demands: PermissionDemands
     let message: String
-
-    public init(type: PermissionType, demands: PermissionDemands, message: String) {
+    
+    let notificationCategories: Set<UIUserNotificationCategory>?
+    init(type: PermissionType, demands: PermissionDemands, message: String, notificationCategories: Set<UIUserNotificationCategory>?) {
```

Comment:
This need to be a public initializer. Also give the notificationCategories a default value of nil so we're not required to say `notificationCategories: nil` in each new config.